cmd/shimdiag/share.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Microsoft/hcsshim/internal/appargs"
+	"github.com/Microsoft/hcsshim/internal/shimdiag"
+	"github.com/urfave/cli"
+)
+
+// LCOW root FS VHD is mounted read only so the only place you can mount a directory
+// is over an existing mount point like /home etc. Single file mapping will not work and
+// will return 'mkdir /path: read-only file system'. You can set the preferred root fs type
+// to initrd to circumvent this.
+
+var readOnly bool
+var shareCommand = cli.Command{
+	Name:      "share",
+	Usage:     "Share a file/directory in a shim's hosting utility VM",
+	ArgsUsage: "<shim name> <host_path> <uvm_path>",
+	Flags: []cli.Flag{
+		cli.BoolFlag{
+			Name:        "readonly,ro",
+			Usage:       "Make the directory/file being shared read only",
+			Destination: &readOnly,
+		},
+	},
+	Before: appargs.Validate(appargs.String, appargs.String, appargs.String),
+	Action: func(c *cli.Context) error {
+		args := c.Args()
+		var (
+			shimName = args[0]
+			hostPath = args[1]
+			uvmPath  = args[2]
+		)
```

Comment:
Does this need to be wrapped with var?

Reply:
Usually if theres multiple variable declarations like this I reach for var over :=, :=, :=. Unless youre referring to something else than just declaring them this way. Do you mean why not just use args[n] directly?

Reply:
No I was just wondering why you didn't use := I can see that being reasonable for a lot of variables, but I'm not sure if its really necessary for just three variables.

Reply:
Well now there's four with readOnly :D

Reply:
fair, but readOnly is not populated from args

Reply:
Does that matter? I can change it if you want I don't mind either way, I just normally declare >= 3 vars like this.

Reply:
I don't think it really matters

Reply:
Going to resolve this then :P